### PR TITLE
FFC-143931: of new process register your interest requests taki…

### PR DESCRIPTION
**Pull request by marcinmo-kainos**
…ng into account multiple sbis

### CODE REVIEW

File: app/auto-eligibility/register-your-interest/check-eligibility.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      businessEmail,
+      sbiAlreadyRegistered: () => true,
+      isEligible: () => false,
+      businessEmailHasMultipleDistinctSbi: () => false,
+      alreadyOnWaitingList: () => false,
+      hasAccessGranted: () => false
+    }
+  }
+  const eligibleCustomer = customers
+    .filter(customer => customer.business_email === businessEmail)
```
Comment (liamwilson-kainos) [inline app/auto-eligibility/register-your-interest/check-eligibility.js:23]: may need to be careful here on the lowercase bug we have as well

Comment (liamwilson-kainos) [reply app/auto-eligibility/register-your-interest/check-eligibility.js:23]: for example if we change the sequalize query to ensure the where condition is in lowercase, will the resultset return the value back to uppercase ->

can be looked at as part of https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/145631

**Review by liamwilson-kainos — CHANGES_REQUESTED** [review]

File: changelog/db.changelog.xml [hunk 2]
```diff
@@ -2,4 +2,5 @@
 <databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
   <include file="changelog/db.changelog-1.0.xml" />
   <include file="changelog/db.changelog-1.1.xml" />
+  <include file="changelog/db.changelog-1.2.xml" />
```
[the remark this answers is not in the document]
Comment (marcinmo-kainos) [reply changelog/db.changelog.xml:5]: done - added "create table vet_visits_data"

**Review by govTechSatish — DISMISSED** [review]
LGTM

File: app/auto-eligibility/register-your-interest/process-eligible-customer.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  }
+  if (customer.businessEmailHasMultipleDistinctSbi()) {
+    console.log('The customer`s business email has multiple sbi which as of now the system does not support')
+    return await emailNotifier.sendIneligibleApplicationEmail(
+      customer.sbi,
+      customer.crn,
+      customer.businessEmail
+    )
+  }
+  if (customer.alreadyOnWaitingList()) {
```
Comment (liamwilson-kainos) [inline app/auto-eligibility/register-your-interest/process-eligible-customer.js:24]: I think this is caught by the sbiAlreadyRegistered() check

**Review by liamwilson-kainos — CHANGES_REQUESTED** [review]

File: app/auto-eligibility/register-your-interest/process-eligible-customer.js (hunk 3 above)
Comment (marcinmo-kainos) [reply app/auto-eligibility/register-your-interest/process-eligible-customer.js]: done - removed

**Pull request merged**